cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator.cpp

```diff
@@ -0,0 +1,472 @@
+            }
+
+            // Calculate the ramp profile path
+            int i = 1;
+            // 0 <= t <= tb
+            for(; i <= steps_tb ; i++)
+            {
+                array.push_back(pa.getStartValue() + direction * (0.5*accl_max*pow((params_.profile.t_ipo*i),2)));
+            }
+            // tb <= t <= tv
+            for(; i <= (steps_tb + steps_tv); i++)
+            {
+                array.push_back(pa.getStartValue() + direction *(vel_max*(params_.profile.t_ipo*i) - 0.5*pow(vel_max,2)/accl_max));
+            }
+            // tv <= t <= te
+            for(; i <= (steps_tv + steps_tb + steps_te + 1); i++)
+            {
+                array.push_back(pa.getStartValue() + direction * (vel_max * (te-tb) - 0.5*accl_max* pow(te-(i*params_.profile.t_ipo),2)));
+            }
+        }
+        else
+        {
+            ROS_INFO_STREAM("Se = 0");
+
+            array.push_back(pa.getStartValue());
+        }
+    }
+    pa.setArray(array);
+    return true;
+}
+
+inline bool TrajectoryProfileRamp::calculateProfile(std::vector<double> path_matrix[4],
+                                                    const double Se, const double Se_roll, const double Se_pitch, const double Se_yaw,
+                                                    geometry_msgs::Pose start)
+{
+    double accl = params_.profile.accl;
+    double vel  = params_.profile.vel;
+    unsigned int max_steps;
+
+    //Convert to RPY
+    double roll_start, pitch_start, yaw_start;
+    tf::Quaternion q;
+    tf::quaternionMsgToTF(start.orientation, q);
+    tf::Matrix3x3(q).getRPY(roll_start, pitch_start, yaw_start);
+
+    std::vector<double> linear_path, roll_path, pitch_path, yaw_path;
+
+    cob_cartesian_controller::PathArray lin(0, Se, 0, linear_path);
+    cob_cartesian_controller::PathArray roll(1, Se_roll, roll_start, roll_path);
+    cob_cartesian_controller::PathArray pitch(2, Se_pitch, pitch_start, pitch_path);
+    cob_cartesian_controller::PathArray yaw(3, Se_yaw, yaw_start, yaw_path);
+    cob_cartesian_controller::PathMatrix pm(lin,roll,pitch,yaw);
+
+    std::vector<cob_cartesian_controller::PathArray> sortedMatrix = pm.getSortedMatrix();   // Sort the Matrix from the largest Se (0) to the smallest one (3)
+
+    ROS_INFO("Got sorted_matrix.");
+
+    this->getProfileTimings(std::fabs(sortedMatrix[0].getSe()), accl, vel);  // Calculate the velocity profile timings with respect to the largest Se
+
+    ROS_INFO("Got profil timings.");
+
+    // Calculate the paths
+    for(int i=0; i<sortedMatrix.size(); i++)
+    {
+        this->generatePath(sortedMatrix[i]);
+    }
+
+    ROS_INFO("GeneratePath success.");
+
+    // Resize the path vectors
+    for(int i = 0; i < sortedMatrix.size(); i++)
+    {
+        max_steps = std::max((unsigned int)sortedMatrix[i].getArray().size() , max_steps);
+    }
+
+    ROS_INFO("max_steps success.");
+
+    // Re-adjust the Matrix to its originally form. Index 0 to 3
```

Comment:
If you really just sorted the PathMatrix in order to find Se_max, this could be done in a simpler way - without needing to change the order in the first place...then re-ordering would of course become obsolete as well....which would significantly simplify the code!